spec/system/candidate_interface/candidate_applying_spec.rb

```diff
@@ -0,0 +1,111 @@
+require 'rails_helper'
+
+describe 'A candidate applying from Find' do
+  let(:provider_code) { '1AB' }
+  let(:course_code) { '2ABC' }
+  let(:course_name) { 'Biology' }
+
+  shared_examples 'displays basic course information' do
+    it 'sees the apply page' do
+      expect(page).to have_content t('applying.heading')
+    end
+
+    it 'sees their provider code' do
+      expect(page).to have_content provider_code
+    end
+
+    it 'sees their course code' do
+      expect(page).to have_content course_code
+    end
+
+    it 'can apply through UCAS' do
+      expect(page).to have_content t('applying.apply_button')
+    end
+  end
+
+  context 'when Find is available' do
+    before do
+      stub_find_is_up
```

Comment:
This method is nice! There's a bit of scrolling to do when coming across `BAD` `COURSE` further down. Would it make sense to pass the `provider_code` and `course_code` as params here, perhaps slightly renaming the method? Then do it for each course that needs to exist in this context. This would make it really explicit whether a given course exists or not in the fake API.